cucumber-expressions/javascript/src/cucumber_expression.js

```diff
@@ -50,10 +50,11 @@ class CucumberExpression {
       if (!parameter) {
         parameter = parameterTypeRegistry.createAnonymousLookup(s => s)
       }
-      this._parameterTypes.push(parameter)
+        parameter.matchType = match[0].split("")[0] === "{" ? "capture" : "non-capture";
```

Comment:
Here you're mutating state in an object that really ought to be immutable. I think it would be better to simply make `matchType` a local variable.